### PR TITLE
Add pthread cond to r_util

### PR DESCRIPTION
**Pull request by thestr4ng3r**
Preparation for my tasks pr. Windows implementation will come later.

### CODE REVIEW

**Comment by radare** [comment]
this seems like a direct wrap to ptrace. how to do this with the w32api? r_th api should be generic and portable

**Comment by radare** [comment]
cc @mrexodia @xarkes @skuater

**Comment by thestr4ng3r** [comment]
That's what I am planning for later. From skipping through the docs it seems to me like it could be wrapped pretty much the same way: https://msdn.microsoft.com/de-de/library/windows/desktop/ms682052(v=vs.85).aspx

pthread_cond_t will be CONDITION_VARIABLE, 
pthread_cond_signal will be WakeConditionVariable, 
pthread_cond_broadcast will be WakeAllConditionVariable, 
pthread_cond_wait will be SleepConditionVariableCS and so on

**Comment by radare** [comment]
👍

File: libr/include/r_th.h [hunk 1]
```diff
@@ -33,6 +35,10 @@ typedef struct r_th_lock_t {
 	R_TH_LOCK_T lock;
 } RThreadLock;
 
+typedef struct r_th_cond_t {
```
Comment (radare) [inline libr/include/r_th.h]: is a struct needed to just hold 1 value?

Comment (thestr4ng3r) [reply libr/include/r_th.h]: maybe not, but the content is system-dependent, so using the struct is the safest way to always keep it universal.

Comment (radare) [reply libr/include/r_th.h]: A define may work too, like its done for R_TH_LOCK_T or R_TH_T

File: libr/util/thread_cond.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+R_API RThreadCond *r_th_cond_new() {
+	RThreadCond *cond = R_NEW0 (RThreadCond);
+	if (!cond) {
+		return NULL;
+	}
+#if HAVE_PTHREAD
+	if (pthread_cond_init (&cond->cond, NULL) != 0) {
+		free (cond);
+		return NULL;
+	}
```
Comment (radare) [inline libr/util/thread_cond.c]:
```
#else 
#warning r_th_cond not supported for this platform
```

**Comment by radare** [comment]
i think the windows implementation can go in this same pr, unless you need a quick merge for you to test other things

**Comment by radare** [comment]
ping

**Comment by thestr4ng3r** [comment]
I am currently working on the tasks stuff (will be done soon™). I will continue with Windows conds after that.

**Comment by radare** [comment]
cool!

**Comment by radare** [comment]
let me know when you want me to merge this pr

**Pull request merged**